js/cedar.js

```diff
@@ -320,22 +322,36 @@ Cedar.Store.prototype.getVersion = function() {
  * @return <Deferred>
  */
 Cedar.Store.prototype.checkVersion = function() {
-  Cedar.debug("checking version #" + this.getVersion());
-  return $.when($.getJSON(Cedar.config.api + '/queries/status'))
-
-  .then( function(response) {
-    if ( Cedar.store.getVersion() != response.settings.version ) {
-      Cedar.debug('setting version: ' + response.settings.version);
-      Cedar.store.loaded = false;
-      Cedar.store.setVersion(response.settings.version);
-    } else {
-      Cedar.debug("version is up to date");
-      Cedar.store.loaded = true;
-      Cedar.$.trigger("content:loaded");
-    }
+  return this.lockedRequest({
+    path: '/queries/status',
+    success: function(response) {
+      Cedar.debug("checking version #" + this.getVersion());
+      if ( this.getVersion() != response.settings.version ) {
+        Cedar.debug('setting version: ' + response.settings.version);
+        this.loaded = false;
+        this.setVersion(response.settings.version);
+      } else {
+        Cedar.debug("version is up to date");
+        this.loaded = true;
+        Cedar.$.trigger("content:loaded");
+      }
+    }.bind(this)
   });
 }
 
+// Returns an already resolving getJSON request if it matches
+Cedar.Store.prototype.lockedRequest = function(options) {
+  options || (options = {});
+  this.requestCache || (this.requestCache = {});
+
+  var requestKey = JSON.stringify({path: options.path, params: options.params});
+
+  return this.requestCache[requestKey] || (this.requestCache[requestKey] = $
```

Comment:
if a stored value evaluates to false, for example it's '0' or 'false' or something, will it break this logic? maybe better to check if `this.requestCache[requestKey]` is defined rather than falsy?